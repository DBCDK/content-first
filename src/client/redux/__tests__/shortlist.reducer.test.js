import shortListReducer, {SHORTLIST_LOAD_RESPONSE} from '../shortlist.reducer';

const createTestState = () => ({
  expanded: false,
  elements: [],
  isLoading: false,
  pendingMerge: null
});

describe('shortListReducer', () => {

  test('when load response include only local storage elements, pendingMerge is null', () => {
    const state = createTestState();
    const action = {
      type: SHORTLIST_LOAD_RESPONSE,
      localStorageElements: [
        {book: {pid: 'pid1'}},
        {book: {pid: 'pid2'}}
      ]
    };
    const result = shortListReducer(state, action);
    expect(result).toMatchSnapshot();
  });

  test('when load response include elements from local storage and database, pendingMerge with diff is created', () => {
    const state = createTestState();
    const action = {
      type: SHORTLIST_LOAD_RESPONSE,
      localStorageElements: [
        {book: {pid: 'pid1'}},
        {book: {pid: 'pid2'}}
      ],
      databaseElements: [
        {book: {pid: 'pid2'}},
        {book: {pid: 'pid3'}}
      ]
    };
    const result = shortListReducer(state, action);
    expect(result).toMatchSnapshot();
  });
});
