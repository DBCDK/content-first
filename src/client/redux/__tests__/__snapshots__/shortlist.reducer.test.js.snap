// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shortListReducer when load response include elements from local storage and database, pendingMerge with diff is created 1`] = `
Object {
  "elements": Array [
    Object {
      "book": Object {
        "pid": "pid2",
      },
    },
    Object {
      "book": Object {
        "pid": "pid3",
      },
    },
  ],
  "expanded": false,
  "isLoading": false,
  "pendingMerge": Object {
    "databaseElements": Array [
      Object {
        "book": Object {
          "pid": "pid2",
        },
      },
      Object {
        "book": Object {
          "pid": "pid3",
        },
      },
    ],
    "diff": Array [
      Object {
        "book": Object {
          "pid": "pid1",
        },
      },
    ],
    "localStorageElements": Array [
      Object {
        "book": Object {
          "pid": "pid1",
        },
      },
      Object {
        "book": Object {
          "pid": "pid2",
        },
      },
    ],
  },
}
`;

exports[`shortListReducer when load response include only local storage elements, pendingMerge is null 1`] = `
Object {
  "elements": Array [
    Object {
      "book": Object {
        "pid": "pid1",
      },
    },
    Object {
      "book": Object {
        "pid": "pid2",
      },
    },
  ],
  "expanded": false,
  "isLoading": false,
  "pendingMerge": null,
}
`;
